유림/230919/B3584.py

```diff
@@ -0,0 +1,35 @@
+
+
+""" [백준] 골드 4. 가장 가까운 공통 조상 """
+
+import sys
+
+input = sys.stdin.readline
+T = int(input())        # 테스트케이스 개수
+
+def solution():
+
+    N = int(input())        # 트리를 구성하는 노드의 수
+
+    trees = [-1] * (N+1)        # 각 노드의 부모
+    visited = [True] + [False] * (N)   # 노드의 방문 여부
+
+    for _ in range(N-1):
+        a, b = map(int, input().split())
+        trees[b] = a
+
+    node1, node2 = map(int, input().split())
+
+    while not visited[node1]:
+        visited[node1] = True
+        if trees[node1] == -1: break
+        node1 = trees[node1]
```

Comment:
```suggestion
    visited[node1] = True:
    while trees[node1] != -1:
        node1 = trees[node1]
        visited[node1] = True
```

요런 식으로 바꿔볼 수도 있을 것 같아요!

물론 취향 차이인데다 알고리즘 문제를 푸는 거라 더더욱 의미가 떨어지긴 하지만

node1 탐색 조건은 루트까지 모두 탐색하는 거여서 while 조건도 똑같이 맞추면 좋겠다 생각했그등여

Reply:
헉 너무 좋아요 더 깔끔하고 보기도 편하네요 😳 고마워요 👍 짱

Reply:
지금말고 저녁에 merge 해야겠네요 ~~ 🐬 기분 좋구만